### PR TITLE
Api lesson update

### PR DESCRIPTION
**Pull request by davidabrahams**

### CODE REVIEW

**Comment by davidabrahams** [comment]
I seperated out the s3 stuff and made updates to my part. I also added a starter android app for the stock app for them to use.

File: lesson7/README.md [hunk 1]
```diff
@@ -1,103 +1,51 @@
...
+You use Volley by creating a `RequestQueue` and passing it `Request` objects. The Queue handles making all of your requests for you, and therefore you should only have one. The Queue will make all your requests and receive their responses.
 
-### Singleton Pattern
-If you are constantly making requests in your app (for example, you make multiple requests to different APIs when your app loads, or whenever the user presses a button a request is made), you will want to use a single instance of a `RequestQueue` that lasts the lifetime of your app. The way you do this is with a singleton class.
+If you are constantly making requests in your app (for example, you make multiple requests to different APIs when your app loads), you will want to use a single instance of a `RequestQueue` that lasts the lifetime of your app. The way you do this is with a singleton class.
 
 The `RequestQueue` must be instantiated in your Application as opposed to your Activity. This is so the queue does not get recreated when the activity is (such as when the device is rotated).
```
Comment (shrutiyer) [inline lesson7/README.md:36]: when the activity is ...? deleted and recreated?

File: lesson7/README.md [hunk 2]
```diff
@@ -119,67 +67,122 @@ public class MySingleton {
...
+
+Programs run one line at a time. If you have a single expression that is slow, it can cause your app to lag. This is generally not an issue, most statements (like `int i = 1 + 1`) are *really* fast. However, you don't know how fast/slow an API will respond. If your program waited for APIs to respond this would be problematic.
+
+One more important aspect of Volley: **you don't have to specify for requests to be done [asynchronously](http://stackoverflow.com/questions/3393751/what-does-asynchronous-means-in-ajax)!** Volley handles all requests on a separate thread and deals with multi-threading for you, and returns all responses to the main thread. Rather than waiting for a request to execute code, the response listener gets called whenever the API responds. Additionally, you can put several requests in the queue and they will all be made one after another, without waiting for each to respond before making the next request.
```
Comment (shrutiyer) [inline lesson7/README.md:157]: Are we linking resources to make async volley calls?

**Review by shrutiyer — APPROVED** [review]
Fix it then ship it?
Looks good! :+1: :stars: :last_quarter_moon_with_face: :sunglasses:

**Comment by nmohamed** [comment]
im good to merge :balloon:

File: lesson7/README.md (hunk 2 above)
Comment (davidabrahams) [reply lesson7/README.md:157]: they're async by default?

File: lesson7/README.md (hunk 1 above)
Comment (davidabrahams) [reply lesson7/README.md:36]: queue is not recreated with the activity is...like, the "recreated" is implicit on the end

**Pull request merged**